backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java

```diff
@@ -77,48 +93,49 @@ private void setup(CveSearchApi cveSearchApi, int vendorThreshold, int productTh
 
     public Stream<List<NeedleWithMeta>> apply(Release release) throws IOException {
         try {
-        return searchLevels.stream()
-                .map(searchLevel -> {
-                    try {
-                        return searchLevel.apply(release);
-                    } catch (IOException e) {
-                        throw new UncheckedIOException(e);
-                    }
-                }) ;
+            return searchLevels.stream()
+                    .map(searchLevel -> {
+                        try {
+                            return searchLevel.apply(release);
+                        } catch (IOException e) {
+                            throw new UncheckedIOException(e);
+                        }
+                    });
         } catch (UncheckedIOException ue) {
             throw ue.getIOExceptionCause();
         }
     }
 
-    class UncheckedIOException extends RuntimeException{
+    class UncheckedIOException extends RuntimeException {
         UncheckedIOException(IOException e) {
             super(e);
         }
 
-        IOException getIOExceptionCause(){
+        IOException getIOExceptionCause() {
             return (IOException) getCause();
         }
     }
 
     //==================================================================================================================
     protected String cleanupCPE(String cpe) {
-        if(cpe.startsWith(OLD_CPE_PREFIX)){ // convert cpe2.2 to cpe2.3
-            cpe = cpe.replaceAll("^"+OLD_CPE_PREFIX, CPE_PREFIX)
+        if (cpe.startsWith(OLD_CPE_PREFIX)) { // convert cpe2.2 to cpe2.3
+            cpe = cpe.replaceAll("^" + OLD_CPE_PREFIX, CPE_PREFIX)
                     .replace("::", ":-:")
                     .replace("~-", "~")
-                    .replace("~",  ":-:")
+                    .replace("~", ":-:")
                     .replace("::", ":")
                     .replaceAll("[:-]*$", "");
         }
         return cpe.toLowerCase();
     }
+
     private void addCPESearchLevel() {
         Predicate<Release> isPossible = r -> r.isSetCpeid() && isCpe(r.getCpeid().toLowerCase());
         searchLevels.add(r -> {
-            if(isPossible.test(r)){
+            if (isPossible.test(r)) {
                 return singletonList(new NeedleWithMeta(cleanupCPE(r.getCpeid()), "CPE"));
             }
-            return EMPTY_LIST;
+            return ImmutableList.of();
```

Comment:
Why do you think this is better than `Collections.EMPTY_LIST`? It doesn't look like a necessary change.

Reply:
It avoids an unchecked cast. It is definitely not necessary, but do we have a guideline to make only necessary changes? I am used to work with a Sonar instance running on the code, so I tend to correct the "warnings" when I see them.

Reply:
No, no, I do not suggest to only make the absolutely necessary changes. I only wondered, what moved you to reduce the readability of the code. IMO, `EMPTY_LIST` is much easier to parse at a glance.
What about using `Collections.emptyList()`? Looks like a win-win to me.

Reply:
I changed both occurences to Collections.emptyList()